src/sentry/web/frontend/auth_login.py

```diff
@@ -127,6 +127,11 @@ def handle_basic_auth(self, request, **kwargs):
             # HACK: grab whatever the first backend is and assume it works
             user.backend = settings.AUTHENTICATION_BACKENDS[0]
 
+            if organization:
```

Comment:
This seems like it would allow anyone to join any organization just by authenticating at that organization's login page no?

Reply:
Yes, that's what I tried to bring attention to:

>Since this also has some security implications (open registration for orgs when SSO is not required), it definitely needs a thorough review. @EvanPurkhiser @markstory

That said I think it makes sense: if I make SSO optional _and_ allow registrations then I effectively let anyone to be able to join the org? Maybe we should limit this behavior to `SINGLE_ORGANIZATION` as that's where the issue surfaces: a user registers without an ability to create a new organization or join the existing one. After the registration, even the invite flow becomes impossible leaving the user account in limbo forever.

Reply:
You cant allow open registration without the customer specifically opting in to that flow, and it should only exist in addition to SINGLE_ORGANIZATION. We have a literal setting to forbid registration which is off by default as its a security vulnerability.